### PR TITLE
Citizen Kill Penality also now relys on Barbarian distance from target

### LINKED ISSUE #2519: Suggestion: Happiness during Barbarian raids
When barbarians are attacking colonist have a habit of getting in the way of arrows/shurikens/swords.  Maybe reduce or have no penalty to player kills on colonist while a barbarian attack is underway.
Thanks

### PR DESCRIPTION
**Pull request by Cazsius**
Closes #2519 

# Changes proposed in this pull request:
- If  a barbarian is within 20 blocks of a citizen when dying, it will not subtract happiness, due to accidental killing during a raid.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 1]
```diff
@@ -371,7 +389,19 @@ public void onDeath(@NotNull final DamageSource damageSource)
             {
                 for (final Player player : PermissionUtils.getPlayersWithAtLeastRank(citizenColonyHandler.getColony(), Rank.OFFICER))
                 {
-                    if (player.getID().equals(damageSource.getTrueSource().getUniqueID()))
+                    boolean isBarbarianClose = true;
+                    for(final AbstractEntityBarbarian barbarian : this.getCitizenColonyHandler().getColony().getBarbManager().getHorde()) {
+                        final EntityCitizen citizen = new EntityCitizen(this.getEntityWorld());
+                        if(MathUtils.twoDimDistance(barbarian.getPosition(), citizen.getPosition()) < 21)
```
[the remark this answers is not in the document]
Comment (Cazsius) [reply src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java]: Should be fine.

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 2]
```diff
@@ -371,7 +389,19 @@ public void onDeath(@NotNull final DamageSource damageSource)
             {
                 for (final Player player : PermissionUtils.getPlayersWithAtLeastRank(citizenColonyHandler.getColony(), Rank.OFFICER))
                 {
-                    if (player.getID().equals(damageSource.getTrueSource().getUniqueID()))
+                    boolean isBarbarianClose = true;
+                    for(final AbstractEntityBarbarian barbarian : this.getCitizenColonyHandler().getColony().getBarbManager().getHorde()) {
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java]: Please add newline before and after {

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 3]
```diff
@@ -371,7 +389,19 @@ public void onDeath(@NotNull final DamageSource damageSource)
...
+                        final EntityCitizen citizen = new EntityCitizen(this.getEntityWorld());
+                        if(MathUtils.twoDimDistance(barbarian.getPosition(), citizen.getPosition()) < BARB_DISTANCE_FOR_FREE_DEATH)
+                        {
+                            isBarbarianClose = true;
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java]: if close is true we can exit the loop already

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 4]
```diff
@@ -371,7 +389,19 @@ public void onDeath(@NotNull final DamageSource damageSource)
             {
                 for (final Player player : PermissionUtils.getPlayersWithAtLeastRank(citizenColonyHandler.getColony(), Rank.OFFICER))
                 {
-                    if (player.getID().equals(damageSource.getTrueSource().getUniqueID()))
+                    boolean isBarbarianClose = true;
+                    for(final AbstractEntityBarbarian barbarian : this.getCitizenColonyHandler().getColony().getBarbManager().getHorde()) {
+                        final EntityCitizen citizen = new EntityCitizen(this.getEntityWorld());
+                        if(MathUtils.twoDimDistance(barbarian.getPosition(), citizen.getPosition()) < BARB_DISTANCE_FOR_FREE_DEATH)
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java]: does the 2dim distance return a square number? Then we need to make Barb_distance ^ 2

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 5]
```diff
@@ -371,7 +389,19 @@ public void onDeath(@NotNull final DamageSource damageSource)
...
+                        {
+                            isBarbarianClose = true;
+                        }
+                        else
+                        {
+                            isBarbarianClose = false;
+                        }
+                    }
+                    if (player.getID().equals(damageSource.getTrueSource().getUniqueID()) && (!isBarbarianClose))
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java]: no need for the extra brackets

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java (hunk 4 above)
Comment (Cazsius) [reply src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java]: does not

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Pull request merged**